pkgs/by-name/ma/marytts/package.nix

```
@@ -0,0 +1,64 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  # Gradle 8 complains about implicit task dependencies when using `installDist`.
+  # See https://github.com/marytts/marytts/issues/1112
+  gradle_7,
+  makeWrapper,
+  jdk,
+  nixosTests,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "marytts";
+  version = "5.2.1-unstable-2024-10-09";
+
+  src = fetchFromGitHub {
+    owner = "marytts";
+    repo = "marytts";
+    rev = "1c2aaa0751b7cef8ae83216dd78b4c61232b3840";
+    hash = "sha256-jGpsD6IwJ67nDLnulBn8DycXCyowssSnDCkQXBIfOH8=";
+  };
+
+  nativeBuildInputs = [
+    gradle_7
+    makeWrapper
+  ];
+
+  mitmCache = gradle_7.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+
+  # Required for the MITM cache to function
+  __darwinAllowLocalNetworking = true;
+
+  gradleBuildTask = "installDist";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv build/install/source/{lib,user-dictionaries} $out
+
+    makeWrapper ${lib.getExe jdk} $out/bin/marytts-server \
+      --add-flags "-cp \"/var/lib/marytts/lib/*:$out/lib/*\" -Dmary.base=/var/lib/marytts" \
+      --append-flags "marytts.server.Mary"
+
+    # We skip the GUI installer since frankly it is a PITA to get to work with a hardened systemd service,
+    # and the imperative installation paradigm is not ideal either way while using Nix.
+
+    runHook postInstall
+  '';
+
+  passthru.tests.nixos = nixosTests.marytts;
```

```suggestion
  passthru.tests = lib.optionalAttrs stdenv.hostPlatform.isLinux {
    nixos = nixosTests.marytts;
  };
```
Lets make sure this isn't attempted by OfBorg on incompatible platforms

---

I'm pretty sure this is unnecessary as I've never put that around a nixosTest or seen it used before. Shouldn't cause any issues though

```
./pkgs/applications/misc/phoc/default.nix
23-, directoryListingUpdater
24:, nixosTests
25-, testers
--
84-  passthru = {
85:    tests.phosh = nixosTests.phosh;
86-    tests.version = testers.testVersion {

./pkgs/applications/misc/keepassxc/default.nix
40-
41:, nixosTests
42-}:
--
142-
143:  passthru.tests = nixosTests.keepassxc;
144-

./pkgs/applications/audio/mopidy/mopidy.nix
10-  pipewire,
11:  nixosTests,
12-}:
--
57-  passthru.tests = {
58:    inherit (nixosTests) mopidy;
59-  };

./pkgs/applications/audio/sfxr-qt/default.nix
11-, callPackage
12:, nixosTests
13-}:
--
50-    export-square-wave = callPackage ./test-export-square-wave {};
51:    sfxr-qt-starts = nixosTests.sfxr-qt;
52-  };

./pkgs/applications/editors/vscode/vscodium.nix
1:{ lib, stdenv, callPackage, fetchurl, nixosTests, commandLineArgs ? "", useVSCodeRipgrep ? stdenv.hostPlatform.isDarwin }:
2-
--
43-
44:    tests = nixosTests.vscodium;
45-

./pkgs/applications/audio/musescore/default.nix
32-, qttools
33:, nixosTests
34-, apple-sdk_11
--
159-
160:  passthru.tests = nixosTests.musescore;
161-

./pkgs/applications/editors/vscode/vscode.nix
5-, fetchurl
6:, nixosTests
7-, srcOnly
--
76-
77:    tests = { inherit (nixosTests) vscode-remote-ssh; };
78-

./pkgs/applications/kde/konsole.nix
1-{
2:  mkDerivation, lib, nixosTests,
3-  extra-cmake-modules, kdoctools,
--
25-
26:  passthru.tests.test = nixosTests.terminal-emulators.konsole;
27-
./pkgs/development/python-modules/private-gpt/default.nix
27-  # tests
28:  nixosTests,
29-}:
--
97-  passthru.tests = {
98:    inherit (nixosTests) private-gpt;
99-  };

./pkgs/development/python-modules/qtile/default.nix
31-  xkbcommon,
32:  nixosTests,
33-  extraPackages ? [ ],
--
96-  passthru = {
97:    tests.qtile = nixosTests.qtile;
98-    providedSessions = [ "qtile" ];

./pkgs/development/python-modules/recipe-scrapers/default.nix
13-  pythonOlder,
14:  nixosTests,
15-}:
--
54-    tests = {
55:      inherit (nixosTests) mealie tandoor-recipes;
56-    };
```

cc @SuperSandro2000

---

> I'm pretty sure this is unnecessary

To prevent OfBorg from always failing on PRs, it is. [Here's](https://github.com/NixOS/nixpkgs/runs/32917973101) the log of a previous commit in this PR where exactly what I described happened -- and I am sure you will see the same for every package you've listed here

>  I've never put that around a nixosTest or seen it used before

That doesn't mean it isn't a good addition. It *is* also used in a number of places through out the tree. [Here's](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+tests+%3D+lib.optionalAttrs+stdenv.hostPlatform.isLinux&type=code) some from a quick search